flutter_appauth/ios/Classes/FlutterAppAuth.m

```diff
@@ -37,8 +37,49 @@ + (NSMutableDictionary *)processResponses:(OIDTokenResponse*) tokenResponse auth
     return processedResponses;
 }
 
-+ (void)finishWithError:(NSString *)errorCode message:(NSString *)message  result:(FlutterResult)result {
-    result([FlutterError errorWithCode:errorCode message:message details:nil]);
++ (void)finishWithError:(NSString *)errorCode message:(NSString *)message result:(FlutterResult)result error:(NSError * _Nullable)error {
+
+    NSMutableDictionary<NSString *, id> *details = [NSMutableDictionary dictionary];
+
+    if (error) {
+        id authError = error.userInfo[OIDOAuthErrorResponseErrorKey];
+        NSDictionary<NSString *, id> *authErrorMap = [authError isKindOfClass:[NSDictionary class]] ? authError : nil;
+        
+        if (authErrorMap) {
+            if ([authErrorMap objectForKey:@"error"]) {
+                [details setObject:authErrorMap[@"error"] forKey:@"error"];
+            }
+            if ([authErrorMap objectForKey:@"error_description"]) {
+                [details setObject:authErrorMap[@"error_description"] forKey:@"error_description"];
+            }
+        }
+        if (error.domain) {
+            [details setObject:error.domain forKey:@"type"];
+        }
+        if (error.code) {
+            [details setObject:[@(error.code) stringValue] forKey:@"code"];
+        }
+        
+        id underlyingErr = [error.userInfo objectForKey:NSUnderlyingErrorKey];
+        NSError *underlyingError = [underlyingErr isKindOfClass:[NSError class]] ? underlyingErr : nil;
+        if (underlyingError) {
+            if (underlyingError.domain) {
+                [details setObject:underlyingError.domain forKey:@"error_uri"];
+            }
+
+            if (underlyingError.debugDescription) {
+                [details setObject:underlyingError.debugDescription forKey:@"root_cause_error"];
+            }
+        } else if (error.debugDescription) {
+            [details setObject:error.debugDescription forKey:@"root_cause_error"];
```

Comment:
From what I could tell, this is a part that could result in confusion. If there's no cause, the text representation of the error is being mapped to a property that represents the cause that would be non-existent. On Android, it will only ever hold information relevant to the cause.

This makes me think in order to achieve what you were trying to do in capturing as much information as possible, the PR would be in need of two properties

1. `rootCauseError`: this what you already have but it should only have populated if the cause exists. On iOS/macOS, it comes from `underlyingError.debugDescription`. On Android, it captures the result of calling `toString()` on the cause of `AuthorizationException`
2. A new property. On iOS, it comes from the `debugDescription` of  the `error` variable. Android, it represents calling `toString()` on the `AuthorizationException`. I can't think of a name for this property 

The intent here is to ensure the information consistently comes from the same level and the name of the property on the Dart side reflects this. Does that make sense?

Reply:
Done :) I've split these into two properties